src/pyhf/contrib/viz/brazil.py

```diff
@@ -1,9 +1,25 @@
 """Brazil Band Plots."""
 import numpy as np
+from collections import namedtuple
 
+BrazilBandArtists = namedtuple(
+    "BrazilBandArtists",
+    ["cls_obs", "cls_exp", "cls_exp_band", "test_size"],
+    defaults=[None, [None] * 5, [None, None], None],
+)
+r"""
+:obj:`collections.namedtuple` of :obj:`matplotlib.artist` in the Brazil band plot.
```

Comment:
For reasons I do not understand, Sphinx is having a fit and failing because of `namedtuple`.

```pytb
WARNING: Failed to get a method signature for pyhf.contrib.viz.brazil.BrazilBandArtists._field_defaults: {'cls_obs': None, 'cls_exp': [None, None, None, None, None], 'cls_exp_band': [None, None], 'test_size': None} is not a callable object
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._field_defaults:3: WARNING: Unexpected indentation.
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._field_defaults:4: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._field_defaults:7: WARNING: Unexpected indentation.
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._field_defaults:8: WARNING: Inline strong start-string without end-string.
WARNING: Failed to get a method signature for pyhf.contrib.viz.brazil.BrazilBandArtists._fields: ('cls_obs', 'cls_exp', 'cls_exp_band', 'test_size') is not a callable object
WARNING: Failed to get a method signature for pyhf.contrib.viz.brazil.BrazilBandArtists._fields_defaults: {'cls_obs': None, 'cls_exp': [None, None, None, None, None], 'cls_exp_band': [None, None], 'test_size': None} is not a callable object
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._fields_defaults:3: WARNING: Unexpected indentation.
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._fields_defaults:4: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._fields_defaults:7: WARNING: Unexpected indentation.
/home/feickert/Code/GitHub/pyhf/src/pyhf/contrib/viz/brazil.py:docstring of pyhf.contrib.viz.brazil.BrazilBandArtists._fields_defaults:8: WARNING: Inline strong start-string without end-string.
/home/feickert/Code/GitHub/pyhf/docs/examples/notebooks/sbottom-pallet/README.md:27: WARNING: Inline literal start-string without end-string.
/home/feickert/Code/GitHub/pyhf/docs/examples/notebooks/sbottom-pallet/README.md:32: WARNING: Inline literal start-string without end-string.
/home/feickert/Code/GitHub/pyhf/docs/examples/notebooks/sbottom-pallet/README.md:32: WARNING: Inline interpreted text or phrase reference start-string without end-string.
```

I haven't figured out how to get it to keep from balking, but if people have ideas that would be great @lukasheinrich @kratsg

Reply:
I do not know what causes this, but have been using `typing.NamedTuple` instead of `collections. namedtuple` myself and not seen such an issue.

Reply:
Hm, okay good to know. I tried this with `typing.NameTuple` already in https://github.com/scikit-hep/pyhf/pull/1377/commits/b08a42760e83dc81241ddff112b438d6febcb93f, but that failed as well.

Reply:
maybe actually it's better to make this a subclass of `matplotlib.container.Container` which "are classes that collect semantically related Artists" 

https://matplotlib.org/stable/api/container_api.html#matplotlib.container.Container

Reply:
> maybe actually it's better to make this a subclass of `matplotlib.container.Container` which "are classes that collect semantically related Artists"
> 
> https://matplotlib.org/stable/api/container_api.html#matplotlib.container.Container

This worked really nicely. Thanks for the great suggestion @lukasheinrich. :+1: 

> and I wonder whether it's a good idea to have an `.axes` property - though e.g. `BarContainer` doesn't have it. But forcing a user to index into some artist such to get the axes seems unsatisfactory.

I think if we want to keep to the `matplotlib` API to make it easy for people to know how to interact with these objects we should avoid it. Also with the current API this isn't too bad

```python
fig = Figure()
ax = fig.subplots()
artists = pyhf.contrib.viz.brazil.plot_results(ax, poi_vals, results)
print(artists.cls_obs.axes)  # AxesSubplot(0.125,0.11;0.775x0.77)
```

Reply:
Might be a good idea for mplhep / hist! :)

Reply:
@henryiii do you have thoughts on the `.axes` property on the container. @matthewfeickert I agree `artists.cls_obs.axes` isn't bad but it's not clear to me why the user needs to think about the sub-artiists if all they want is the axes. I could imagine the reason `.axes` is not part of the  `Container` base class is thatt in princicple a container could collect artists attached to various axes instances

Reply:
Added for now in https://github.com/scikit-hep/pyhf/pull/1377/commits/a3e8ec6206e376df5e6e8566b4f3312f22f3a14b